### PR TITLE
airbyte-server: set tags on span in `OAuthHandler`

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-cloud/issues/3672
To track OAuth errors per connector, we must enrich our traces with new tags.

## How
Add tags to traces in the following methods of `OAuthHandler`:
* `getSourceOAuthConsent`
* `getDestinationOAuthConsent`
* `completeSourceOAuth`
* `completeDestinationOAuth`

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

File: airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java [hunk 1]
```diff
@@ -195,6 +208,9 @@ public Map<String, Object> completeSourceOAuth(final CompleteSourceOauthRequest
 
   public Map<String, Object> completeDestinationOAuth(final CompleteDestinationOAuthRequest completeDestinationOAuthRequest)
       throws JsonValidationException, ConfigNotFoundException, IOException {
+    ApmTraceUtils.addTagsToTrace(Map.of(WORKSPACE_ID_KEY, completeDestinationOAuthRequest.getWorkspaceId(), DESTINATION_DEFINITION_ID_KEY,
+        completeDestinationOAuthRequest.getSourceDefinitionId()));
```
Comment (pedroslopez) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java:212]:
```suggestion
        completeDestinationOAuthRequest.getDestinationDefinitionId()));
```

Is this a typo or is this name actually wrong in the class?

**Review by pedroslopez — CHANGES_REQUESTED** [review]
Amazing!

File: airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java (hunk 1 above)
Comment (alafanechere) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java:212]: Typo! Thanks for catching it

**Review by pedroslopez — APPROVED** [review]

**Pull request merged**